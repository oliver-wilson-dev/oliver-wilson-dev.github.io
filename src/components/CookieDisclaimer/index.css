:root {
    --cookie-disclaimer-text-colour: #333;
    --cookie-disclaimer-text-font-size: 16px;
}