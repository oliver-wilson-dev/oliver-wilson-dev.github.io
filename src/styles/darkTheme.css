@import "./index.css";

[data-theme="DARK_THEME"] {
    --primaryBackground: #333;
    --primaryBackgroundImage: linear-gradient(#333, #333);
    --primaryBorderColour: #f8f8f8;
    --primaryTextColor: #f8f8f8;
    --spinnerTrailColour: rgba(248,248,248, 0.2);
    --sectionFadeColourFrom: rgba(51, 51, 51, 0);
    --sectionFadeColourTo: rgba(51, 51, 51, 1);
    --unvisitedLink: var(--projectGreen)
}