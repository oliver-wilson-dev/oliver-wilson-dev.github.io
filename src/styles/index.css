:root {
    --primaryBackgroundLight: #f8f8f8;
    --primaryBackgroundDark: #333;
    --primaryBackground: var(--primaryBackgroundLight);

    --primaryBorderColour: #0b0a0a;
    --primaryBorderColourLight: #0b0a0a;

    --primaryTextColor: #333;

    --primaryTextColorDark: #f8f8f8;
    --primaryTextColorLight: #333;

    --primaryTextColorInverse: var(--primaryTextColorLight);

    --spinnerTrailColour: rgba(11,10,10, 0.2);
    --sectionFadeColourFrom: rgba(248, 248, 248, 0);
    --sectionFadeColourTo: rgba(248, 248, 248, 1);

    --linkBlue: rgb(21, 64, 204);
    --unvisitedLink: var(--linkBlue);

    --profilePictureBorder: var(--primaryBorderColour);

    --headerHeight: 65px;
    --headerPadding: 15px;

    --overlayBackground: rgba(0,0,0,.288);

    --projectGreen: #06d6a0;
    --projectPink: #EF476F;
    --projectYellow: #FFD166;
    --projectBlue: #118AB2;
    --projectDarkBlue: #073B4C;
}