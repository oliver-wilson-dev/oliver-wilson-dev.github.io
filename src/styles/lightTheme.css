@import "./index.css";

[data-theme="LIGHT_THEME"] {
    --primaryBackground: var(--primaryBackgroundLight);
    --primaryBackgroundInverse: var(--primaryBackgroundDark);

    --primaryBorderColour: #0b0a0a;
    --primaryBorderColourLight: #0b0a0a;

    --primaryTextColor: var(--primaryTextColorLight);
    --primaryTextColorInverse: var(--primaryTextColorDark);

    --spinnerTrailColour: rgba(11,10,10, 0.2);
    --sectionFadeColourFrom: rgba(248, 248, 248, 0);
    --sectionFadeColourTo: rgba(248, 248, 248, 1);

    --primaryBackgroundLight: #f8f8f8;

    --unvisitedLink: var(--projectGreen);
    --unvisitedLinkInverse: var(--projectGreen);
}