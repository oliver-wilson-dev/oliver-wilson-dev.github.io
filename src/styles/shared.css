.bodyFixed {
    position: fixed;
    left: 0;
    width: 100%;
    overflow-y: scroll;
}

.pageHeader {
    font-size: 5em;
    margin: 0;
    margin-top: 10px;
    background-clip: text;
    -webkit-background-clip: text;
    letter-spacing: 2px;
    text-align: center;
    width: 100%;
  }

  @media (min-width: 786px) {
    .landing--header {
    max-width: 800px;
    }
  }

  @media (min-width: 0px) and (max-width: 767px) {
    .pageHeader {
      font-size: 3em;
    }
  }