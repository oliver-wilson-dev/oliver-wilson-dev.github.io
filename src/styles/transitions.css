:root {
    --spinnerOpacityTransitionDuration: 1s;
    --sectionOpacityTransitionDuration: 0.5s;
    --sectionVisibilityTransitionDuration: 0s;
}