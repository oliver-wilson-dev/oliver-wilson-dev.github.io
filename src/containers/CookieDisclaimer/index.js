import { connect } from 'react-redux';
import CookieDisclaimer from '../../components/CookieDisclaimer';
import { getShowLearnMore } from '../../state/selectors';

const mapStateToProps = state => ({
  showLearnMore: getShowLearnMore(state),
});

export default connect(mapStateToProps)(CookieDisclaimer);
