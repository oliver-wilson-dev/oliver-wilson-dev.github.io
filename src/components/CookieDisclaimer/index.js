import React, { useState, useEffect } from 'react';
import cn from 'classnames';
import styles from './index.css';

const CookieDisclaimer = () => {
  const [acknowledged, setAcknowledged] = useState(false);
  const [loaded, setLoaded] = useState(false);
  const [closed, setClosed] = useState(false);

  useEffect(() => {
    setLoaded(true);
  }, []);

  const onTransitionEnd = () => {
    if (acknowledged) setClosed(true);
  };

  const acknowledgeBanner = () => {
    setAcknowledged(true);
  };

  return !closed && (
    <div
      className={cn(styles.container, {
        [styles.loaded]: loaded && !acknowledged,
        [styles.closed]: acknowledged
      })}
      onTransitionEnd={onTransitionEnd}
    >
      <div className={styles.positionWrapper}>
        <div className={cn(styles.cookieEmoji, styles.notificationElement)} role="img" aria-label="cookie emoji" />
        <span className={cn(
          styles.message,
          styles.notificationElement,
        )}
        >
          Attention! This site uses cookies.
        </span>
        <button className={cn(styles.tick, styles.notificationElement)} type="button" onClick={acknowledgeBanner} />
      </div>
    </div>
  );
};

export default CookieDisclaimer;
