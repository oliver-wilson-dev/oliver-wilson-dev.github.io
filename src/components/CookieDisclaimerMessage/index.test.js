import React from 'react';
import { shallow, mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import styles from './index.css';

import CookieDisclaimer from '.';

const defaultProps = {
  learnMoreClicked: jest.fn()
};

const render = (renderMethod = shallow, props = {}) => renderMethod(
  <CookieDisclaimer
    {...defaultProps}
    {...props}
  />
);

describe('Cookie Disclaimer component', () => {
  describe('when first loaded', () => {
    it('should render correctly', () => {
      expect(render()).toMatchSnapshot();
    });

    it('should then apply the loaded class', () => {
      expect(render(mount)).toMatchSnapshot();
    });

    describe('and the initial animation happens', () => {
      it('should not close', () => {
        const component = render(mount);
        const container = component.find(`.${styles.container}`);
        container.props().onTransitionEnd();

        expect(container.hasClass(styles.loaded)).toBe(true);
        expect(container.hasClass(styles.closed)).toBe(false);
      });
    });

    describe('when the user accepts the policy', () => {
      it('should render correctly', () => {
        const component = render(mount);

        component.find(`.${styles.tick}`).simulate('click');

        expect(component).toMatchSnapshot();
      });

      describe('after the transition ends', () => {
        it('should render null', () => {
          const component = render();

          component.find(`.${styles.tick}`).simulate('click');

          act(() => {
            /* fire events that update state */
            component.find(`.${styles.closed}`).props().onTransitionEnd();
          });

          component.update();

          expect(component).toMatchSnapshot();
        });
      });
    });

    describe('when the user clicks learn more', () => {
      it('should render correctly', () => {
        const component = render(mount);

        component.find(`.${styles.learnMoreBtn}`).simulate('click');

        expect(component).toMatchSnapshot();
      });

      it('should call the learnMoreClickedProp', () => {
        const props = {
          learnMoreClicked: jest.fn()
        };

        const component = render(mount, props);

        component.find(`.${styles.learnMoreBtn}`).simulate('click');

        expect(props.learnMoreClicked).toHaveBeenCalledTimes(1);
      });

      describe('after the transition ends', () => {
        it('should render null', () => {
          const component = render();

          component.find(`.${styles.learnMoreBtn}`).simulate('click');

          act(() => {
            /* fire events that update state */
            component.find(`.${styles.closed}`).props().onTransitionEnd();
          });

          component.update();

          expect(component).toMatchSnapshot();
        });
      });
    });
  });
});
