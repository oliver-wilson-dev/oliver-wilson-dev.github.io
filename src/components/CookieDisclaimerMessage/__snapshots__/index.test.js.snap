// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cookie Disclaimer component when first loaded should render correctly 1`] = `
<div
  className="container"
  onTransitionEnd={[Function]}
>
  <div
    className="positionWrapper"
  >
    <div
      aria-label="cookie emoji"
      className="cookieEmoji notificationElement"
      role="img"
    />
    <span
      className="message text notificationElement"
    >
      Attention! This site uses cookies.
       
      <button
        className="learnMoreBtn text"
        onClick={[Function]}
        type="button"
      >
        Learn More
      </button>
    </span>
    <button
      className="tick notificationElement"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;

exports[`Cookie Disclaimer component when first loaded should then apply the loaded class 1`] = `
<CookieDisclaimerMessage
  learnMoreClicked={[MockFunction]}
>
  <div
    className="container loaded"
    onTransitionEnd={[Function]}
  >
    <div
      className="positionWrapper"
    >
      <div
        aria-label="cookie emoji"
        className="cookieEmoji notificationElement"
        role="img"
      />
      <span
        className="message text notificationElement"
      >
        Attention! This site uses cookies.
         
        <button
          className="learnMoreBtn text"
          onClick={[Function]}
          type="button"
        >
          Learn More
        </button>
      </span>
      <button
        className="tick notificationElement"
        onClick={[Function]}
        type="button"
      />
    </div>
  </div>
</CookieDisclaimerMessage>
`;

exports[`Cookie Disclaimer component when first loaded when the user accepts the policy after the transition ends should render null 1`] = `""`;

exports[`Cookie Disclaimer component when first loaded when the user accepts the policy should render correctly 1`] = `
<CookieDisclaimerMessage
  learnMoreClicked={[MockFunction]}
>
  <div
    className="container closed"
    onTransitionEnd={[Function]}
  >
    <div
      className="positionWrapper"
    >
      <div
        aria-label="cookie emoji"
        className="cookieEmoji notificationElement"
        role="img"
      />
      <span
        className="message text notificationElement"
      >
        Attention! This site uses cookies.
         
        <button
          className="learnMoreBtn text"
          onClick={[Function]}
          type="button"
        >
          Learn More
        </button>
      </span>
      <button
        className="tick notificationElement"
        onClick={[Function]}
        type="button"
      />
    </div>
  </div>
</CookieDisclaimerMessage>
`;

exports[`Cookie Disclaimer component when first loaded when the user clicks learn more after the transition ends should render null 1`] = `""`;

exports[`Cookie Disclaimer component when first loaded when the user clicks learn more should render correctly 1`] = `
<CookieDisclaimerMessage
  learnMoreClicked={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <div
    className="container closed"
    onTransitionEnd={[Function]}
  >
    <div
      className="positionWrapper"
    >
      <div
        aria-label="cookie emoji"
        className="cookieEmoji notificationElement"
        role="img"
      />
      <span
        className="message text notificationElement"
      >
        Attention! This site uses cookies.
         
        <button
          className="learnMoreBtn text"
          onClick={[Function]}
          type="button"
        >
          Learn More
        </button>
      </span>
      <button
        className="tick notificationElement"
        onClick={[Function]}
        type="button"
      />
    </div>
  </div>
</CookieDisclaimerMessage>
`;
